libs/muddle/src/promise_runnable.cpp

```diff
@@ -20,16 +20,78 @@
 
 namespace fetch {
 namespace muddle {
+namespace {
 
-static constexpr char const *LOGGING_NAME = "PromiseTask";
+constexpr char const *LOGGING_NAME = "PromiseTask";
 
-PromiseTask::PromiseTask(service::Promise promise, Callback callback)
-  : promise_{std::move(promise)}
+using Clock        = PromiseTask::Clock;
+using Timepoint    = PromiseTask::Timepoint;
+using Duration     = PromiseTask::Duration;
+using Promise      = service::Promise;
+using PromiseState = service::PromiseState;
+
+/**
+ * Compute the desired timeout or deadline for the monitored promise
+ *
+ * @param promise The promise to be monitored
+ * @param timeout The (optional) timeout to be imposed
+ * @return The desired deadline time
+ */
+Timepoint CalculateDeadline(Promise const &promise, Duration const *timeout = nullptr)
```

Comment:
Why not simply split it in two overloads rather than pass optional arg as pointer?

Reply:
Hmmm, honestly I was thinking that is easier to read the intention this way. I have changed it let me know what you think 😁

Reply:
It's better this way better IMO, two one-liners against one function with default arg values and branches. It would even be shorter if not for duplicated doc comment. >_<